### PR TITLE
Aatrox buff calculation in Slayer Overlay

### PR DESCRIPTION
**Pull request by Taoshix**
If Aatrox has his +25% xp buff active, update bosses to next level calculation to use the increased gains

### CODE REVIEW

**Review by jani270 — CHANGES_REQUESTED** [review]
Two things:
1. Please format your code.
2. Please add yourself to the 2.1.1.md

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java [hunk 1]
```diff
@@ -117,6 +119,22 @@ public void update() {
 			isSlayerNine = true;
 		} else if (!slayerXp.equals("0")) {
 			slayerEXP = slayerXp.replace(",", "");
+
+			differenceFromLastXP = slayerIntXP-Integer.parseInt(slayerEXP);
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java:123]: Hmmm

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java [hunk 2]
```diff
@@ -117,6 +119,22 @@ public void update() {
...
+			if(differenceFromLastXP != 0){
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java:124]: Please reformat your code correctly. What IDE do you use? When Intellij, just press control +alt + L. If you use VSC or something else, just manually add spaces

**Review by hannibal002 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java (hunk 2 above)
Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java:124]: grafik

Comment (Taoshix) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java]: > Please reformat your code correctly. What IDE do you use? When Intellij, just press control +alt + L. If you use VSC or something else, just manually add spaces

I do use Intellij and I downloaded it a few days ago so I am not that familiar with it yet, thank you for showing me the reformat keybind.

Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/SlayerOverlay.java]: Happy to help :)

**Review by hannibal002 — APPROVED** [review]
Strangely, this calculation matches Hypixel's own logic. The +25% buff for T1 (+5 XP) is in reality only a 20% buff (+5 XP → +6 XP). Meaning, this PR makes the same mistake as Hypixel does, and it's fine.

**Comment by Taoshix** [comment]
> Strangely, this calculation matches Hypixel's own logic. The +25% buff for T1 (+5 XP) is in reality only a 20% buff (+5 XP → +6 XP). Meaning, this PR makes the same mistake as Hypixel does, and it's fine.

I was wondering if they on their backend store the xp value as a float so it will roll over every 4th boss to be +7 xp.
Since I don't know I just assumed that they round the number down every time

**Comment by hannibal002** [comment]
> Since I don't know if they do that or not I just assumed that they round the number down every time

The API can confirm that your assumption is correct. They don't give you more exp every x bosses.
grafik

**Comment by Taoshix** [comment]
I found a bug where if you need less xp to level up than what you would gain, it would say that you need 0 bosses instead of 1 boss

**Comment by Mqisty** [comment]
> I found a bug where if you need less xp to level up than what you would gain, it would say that you need [0 bosses instead of 1](https://cdn.discordapp.com/attachments/494861267576815616/1041024757824495736/image.png) boss.

i reported this a while ago but no one cares enough to fix it

**Comment by Taoshix** [comment]
> > I found a bug where if you need less xp to level up than what you would gain, it would say that you need [0 bosses instead of 1](https://cdn.discordapp.com/attachments/494861267576815616/1041024757824495736/image.png) boss.
> 
> i reported this a while ago but no one cares enough to fix it

Fair enough, I just thought that since I was working on the Slayer Overlay already, I might as well fix it

**Review by hannibal002 — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]
looks sane enough

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**